### PR TITLE
feat(ui-text-input,ui-date-input): properly support IconButtons inside TextInputs

### PR DESCRIPTION
**Pull request by balzss**
INSTUI-4614

test plan:

- check new docs and example for TextInput with the iconbutton rendered at the end
- check out dateinput2
- check the code, please try to think about edge cases. I might have forgot about something that could break

### CODE REVIEW

File: packages/ui-text-input/src/TextInput/README.md [hunk 1]
```diff
@@ -301,6 +298,47 @@ Focusable content will be focused separately from the input itself.
...
+---
+type: example
+---
+<TextInput
+  renderLabel="Search"
+  type="search"
+  placeholder="Find something..."
+  renderBeforeInput={<IconSearchSolid/>}
+/>
+```
```
Comment (matyasf) [inline packages/ui-text-input/src/TextInput/README.md]: isnt it an a11y problem that the close button here does not have a focus ring and is not in the keyboard tab order?

File: packages/ui-text-input/src/TextInput/README.md [hunk 2]
```diff
@@ -301,6 +298,47 @@ Focusable content will be focused separately from the input itself.
...
+    <TextInput
+      renderLabel="Password"
+      type={showPassword ? 'text' : 'password'}
+      placeholder="Find something..."
+      value={passwordValue}
+      onChange={(e, newValue) => setPasswordValue(newValue)}
+      renderAfterInput={
+        <IconButton withBorder={false} withBackground={false} onClick={() => setShowPassword(prevState => !prevState)}>
+          {showPassword ? <IconOffLine/> : <IconEyeLine/>}
+        </IconButton>
```
Comment (matyasf) [inline packages/ui-text-input/src/TextInput/README.md]: SRs say here just Button, please add some text for them, something like "show/hide text"

File: packages/ui-text-input/src/TextInput/README.md [hunk 3]
```diff
@@ -301,6 +298,47 @@ Focusable content will be focused separately from the input itself.
...
+      }
+    />
+  )
+}
+render(<InputsWithButtonsExample />)
+```
+
+Having a clear button for a search field is also common but you don't need to implement anything to make that work, only set the `type` of the input to `search`:
```
Comment (matyasf) [inline packages/ui-text-input/src/TextInput/README.md]: I dont see a close button in FF for this example.

**Review by matyasf — CHANGES_REQUESTED** [review]
see my comments. Also

File: packages/ui-text-input/src/TextInput/README.md (hunk 3 above)
Comment (balzss) [reply packages/ui-text-input/src/TextInput/README.md]: tbh I wasn't really sure where that feature was coming from but seems like it's a chrome thing. I've added this example to show people that they don't need the iconbutton when it's `type="search"`. I guess this feature is not really a complete solution so I'll remove this example. later on we could discuss what to do/suggest with `type=search` input fields

File: packages/ui-text-input/src/TextInput/README.md (hunk 1 above)
Comment (balzss) [reply packages/ui-text-input/src/TextInput/README.md]: see my comment about the search input below

File: packages/ui-text-input/src/TextInput/README.md (hunk 2 above)
Comment (balzss) [reply packages/ui-text-input/src/TextInput/README.md]: done

File: packages/ui-text-input/src/TextInput/README.md (hunk 1 above)
Comment (balzss) [reply packages/ui-text-input/src/TextInput/README.md]: I've removed this example
